source/patterns/@aws-solutions-constructs/core/lib/bedrock-inference-helper.ts

```diff
@@ -0,0 +1,140 @@
+/**
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ *  with the License. A copy of the License is located at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+
+// Imports
+import * as cdk from 'aws-cdk-lib';
+import * as bedrock from 'aws-cdk-lib/aws-bedrock';
+import * as defaults from '../index';
+// Note: To ensure CDKv2 compatibility, keep the import statement for Construct separate
+import { Construct } from 'constructs';
+
+export interface BuildInferenceProfileProps {
+  readonly bedrockModelId: string;
+  readonly inferenceProfileProps?: bedrock.CfnApplicationInferenceProfileProps;
+  readonly deployCrossRegionProfile?: boolean;
+}
+
+export interface BuildInferenceProfileReponse {
+  readonly crossRegion?: boolean,
+  readonly inferenceProfile: bedrock.CfnApplicationInferenceProfile,
+}
+
+export function buildInferenceProfile(scope: Construct, id: string, props: BuildInferenceProfileProps): BuildInferenceProfileReponse {
+  const areaMap = createAreaPrefixMapping(scope, id);
+
```

Comment:
should there be a check somewhere here if both, `bedrockModelId` and `modelSource` are set, to throw an error.